src/logs/fatal_on.go

```diff
@@ -0,0 +1,46 @@
+package logs
+
+import (
+	"log"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// FatalOn runs log.Fatal with the given error
+// when the given error exists.
+// If an optional message is provided,
+// the error is wrapped in it.
+func FatalOn(err error, messages ...string) {
```

Comment:
I'd prefer we have separate methods for no message and with a single message. I don't think we would ever pass in multiple messages. There appears to be a couple places where you are using this method when we should be using `FatalfOn`.

Reply:
This would be nice in Java that allows methods with the same name but different parameter sets. In Go we would have to pick a different method name, and I'm unsure which one makes sense here. 

I actually like the current interface: you can call `log.FatalOn` with an error and an optional text message. If no message is given, it just logs the error. If a message is given, it logs the error wrapped in the message. 

The additional type safety for preventing multiple messages doesn't seem worth the extra complexity, given that if multiple messages are given we simply concatenate them. What types of errors do you expect to prevent with this? 

Also, this matches the behavior of `log.Fatal` and `fmt.Print`. We want to stay close to those.

Reply:
I would prefer we mirror the `errors.Wrap` API more than the `log.Fatal`. We are combining those two APIs and I think wrapping the error is more important.

As I displayed in a comment earlier, I would prefer an API:
```go
logs.FatalOn(err)
logs.FatalOnWrap(err, message) // equivalent to log.FatalOn(errors.Wrap(err, message))
logs.FatalOnWrapf(err, format, v ...interface{}) // equivalent to logs.FatalOn(errors.Wrapf(err, format, ...v)
```

Thus we are basically only introducing one new method `logs.FatalOn` and then building two methods that build on top of it